Relazione/RelazioneTecWeb.tex

```diff
@@ -125,26 +125,31 @@
 			\subsubsection{CSS}
 				CSS è stato usato per la presentazione grafica delle pagine HTML. Si è cercato di puntare ad un discreto riutilizzo del codice creando classi da poter usare su più pagine, come per esempio bottoni e form preimpostati, che poi potevano venire personalizzati dove necessario sovrascrivendo le regole necessarie. Il contenuto di ogni pagina è identificato da una classe così da avere una formattazione del testo omogenea mentre per impaginare le pagine specifiche si è optato per un identificativo univoco per ogni pagina.
 			\subsubsection{PHP}
+			Il linguaggio PHP è stato utilizzato per creare tutte le interazioni del sito con il database, oltre che ad effettuare dei controlli lato server per quanto riguarda la validità e la bontà degli input forniti dagli utenti. Un altro uso che è stato fatto di questo linguaggio è stato quello di creare delle pagine web dinamiche che avessero un menù personalizzato, in base al fatto che un utente fosse autenticato o meno, e dei contenuti diversi in base ai dati presenti nel database (e.g. le destinazioni salvate o le notizie pubblicate dagli amministratori).
+			Per quanto riguarda la codifica si è cercato, ove possibile, di definire delle funzioni che realizzassero i compiti che più si sarebbero ripetuti all'interno del sito, come il controllo degli input, e generalmente si è cercato di separare la logica della pagina dalle funzioni chiamate in modo tale da diminuire il più possibile le dipendenze ed agevolare un'eventuale manutenzione del sito.
 			\subsubsection{SQL}
 			Sql è stato usato per codificare il database. Si rimanda al file \textit{creazione\_database.sql} nella cartella \textit{Database} [\href{https://github.com/Mirco469/ProgettoSushi/tree/master/Database}{url}] della repository per il file di costruzione del database. Di seguito il diagramma ER del database:\newline
 			\includegraphics[width=12cm]{DiagrammaER.png}
 			\subsubsection{JavaScript}
+			Il linguaggio JavaScript è stato utilizzato per lo più per fare un controllo preventivo dei campi di input, senza richiedere la ricarica della pagina per dei semplici controlli tramite delle espressioni regolari. Il linguaggio è stato anche utilizzato per creare uno slideshow a scorrimento manuale nella pagina di inizio dell'utente, con delle immagini che rappresentassero il clima del ristorante, senza però scordarsi dell'accessibilità e del disturbo che potrebbe arrecare un carousel automatico. Un altro utilizzo di JavaScript è stato riguardo la creazione di un menù a scomparsa nella versione mobile del sito in modo tale da sfruttare al massimo la dimensione ridotta delle pagine ed evitare degli scroll verticali aggiuntivi che sarebbero stati richiesti nel caso di un menù verticale non a scomparsa. Come per il linguaggio PHP si è puntato al riutilizzo del codice definendo funzioni per effettuare i controlli e riutilizzando le funzioni dove possibile.
```

Comment:
Piuttosto che dire che usiamo javascript per non ricaricare la pagina che invece lo facciamo lo stesso direi che è per non dover far fare lavoro al server per controlli che possono essere già catturati mentre ci si trova nel client (ovviamente bisogna metterlo con una frase decente)

Reply:
Ha più senso in effetti